ui/window.go

```diff
@@ -1871,6 +1890,49 @@ func (window *Window) SwitchToFriendsPage() {
 	}
 }
 
+// Switches to the previous channel and layout.
+func (window *Window) SwitchToPreviousChannel() {
+	if window.previousChannel == nil || window.previousChannel == window.selectedChannel {
+		return
+	}
+
+	_, err := window.session.State.Guild(window.previousGuild.ID)
+	if err != nil {
+		window.previousGuild = nil
+		window.previousGuildNode = nil
+		return
+	}
+
+	_, err = window.session.State.Channel(window.previousChannel.ID)
+	if err != nil {
+		window.previousChannel = nil
+		window.previousChannelNode = nil
+		return
+	}
+
+	// Switch to appropriate layout.
+	switch window.previousChannel.Type {
+	case discordgo.ChannelTypeDM, discordgo.ChannelTypeGroupDM:
+		if window.leftArea.GetCurrentPage() != privatePageName {
+			window.leftArea.SwitchToPage(privatePageName)
+			window.privateList.onChannelSelect(window.previousChannelNode, window.previousChannel.ID)
+		}
+	default:
```

Comment:
Please replace this with the proper case and return and show an error dialig on default.